pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -0,0 +1,173 @@
+
+  public LLRealtimeSegmentDataManager(RealtimeSegmentZKMetadata segmentZKMetadata, AbstractTableConfig tableConfig,
+      InstanceZKMetadata instanceZKMetadata, RealtimeTableDataManager realtimeTableDataManager, String absolutePath,
+      ReadMode readMode, Schema schema, ServerMetrics serverMetrics) throws Exception {
+    _segmentZKMetadata = (LLCRealtimeSegmentZKMetadata) segmentZKMetadata;
+    _tableConfig = tableConfig;
+    _instanceZKMetadata = instanceZKMetadata;
+    _realtimeTableDataManager = realtimeTableDataManager;
+    _absolutePath = absolutePath;
+    _readMode = readMode;
+    _schema = schema;
+    _serverMetrics = serverMetrics;
+
+    // Load configs
+    // TODO Validate configs
+    final String bootstrapNodes = _tableConfig.getIndexingConfig().getStreamConfigs()
+        .get(CommonConstants.Helix.DataSource.STREAM_PREFIX + "." + CommonConstants.Helix.DataSource.Realtime.Kafka.KAFKA_BROKER_LIST);
+    final String kafkaTopic = _tableConfig.getIndexingConfig().getStreamConfigs()
+        .get(CommonConstants.Helix.DataSource.STREAM_PREFIX + "." + CommonConstants.Helix.DataSource.Realtime.Kafka.TOPIC_NAME);
+    final LLCSegmentName segmentName = new LLCSegmentName(_segmentZKMetadata.getSegmentName());
+    final int kafkaPartitionId = segmentName.getPartitionId();
+    final String decoderClassName = _tableConfig.getIndexingConfig().getStreamConfigs()
+        .get(CommonConstants.Helix.DataSource.STREAM_PREFIX + "." + CommonConstants.Helix.DataSource.Realtime.Kafka.DECODER_CLASS);
+
+    // Start new realtime segment
+    // HACK jfim hardcoded value for segment size
+    _realtimeSegment = new RealtimeSegmentImpl(schema, 200000, tableConfig.getTableName(),
+        segmentZKMetadata.getSegmentName(), kafkaTopic, serverMetrics);
+    _realtimeSegment.setSegmentMetadata(segmentZKMetadata, schema);
+
+    // Create message decoder
+    final KafkaMessageDecoder messageDecoder = (KafkaMessageDecoder) Class.forName(decoderClassName).newInstance();
+    messageDecoder.init(new HashMap<String, String>(), _schema, kafkaTopic);
+
+    // Create field extractor
+    final PlainFieldExtractor fieldExtractor = (PlainFieldExtractor) FieldExtractorFactory.getPlainFieldExtractor(schema);
+
+    // Start indexing thread
+    _indexingThread = new Thread(new Runnable() {
+      @Override
+      public void run() {
+        try {
+          // Create Kafka consumer
+          SimpleConsumerWrapper consumerWrapper =
+              SimpleConsumerWrapper.forPartitionConsumption(bootstrapNodes, kafkaTopic, kafkaPartitionId);
+
+          // TODO Check for limit conditions (eg. full, stop consuming)
+          boolean notFull = true;
+          long currentOffset = _segmentZKMetadata.getStartOffset();
+
+          while (notFull) {
+            // Get a batch of messages from Kafka
+            Iterable<MessageAndOffset> messagesAndOffsets =
+                consumerWrapper.fetchMessages(currentOffset, Long.MAX_VALUE, 1000);
```

Comment:
please declare the 1000 and maxvalue as final with some var names and use them, so it is clear what they stand for.